test/functional/ui/screen.lua

```diff
@@ -1246,6 +1260,26 @@ function Screen:print_snapshot(attrs, ignore)
     end
   end
 
+  -- Build keyword-args for screen:expect().
+  local kwargs = {}
+  if attr_state.modified then
+    kwargs['attr_ids'] = {}
+    for i, a in pairs(attr_state.ids) do
+      kwargs['attr_ids'][i] = a
```

Comment:
Will this work for `self._options.ext_hlstate` also?

Reply:
Not sure, will check later. Some of the differences between hlstate and standard linegrid are just technical debt, I should clean it up..